cellprofiler/modules/fillobjects.py

```diff
@@ -0,0 +1,90 @@
+# coding=utf-8
+
+"""
+FillObjects
+===========
+
+**FillObjects** fills holes within all objects in an image.
+
+This module works best on integer-labeled images (i.e., the output of **ConvertObjectsToImage**
+when the color format is *uint16*).
+
+The output of this module is a labeled image of the same data type as the input.
+**FillObjects** can be run *after* any labeling or segmentation module (e.g.,
+**ConvertImageToObjects** or **Watershed**). Labels are preserved and, where possible, holes
+entirely within the boundary of labeled objects are filled with the surrounding object number.
+
+|
+
+============ ============ ===============
+Supports 2D? Supports 3D? Respects masks?
+============ ============ ===============
+YES          YES          NO
+============ ============ ===============
+
+"""
+
+import numpy
+import skimage.morphology
+
+import cellprofiler.image
+import cellprofiler.module
+import cellprofiler.setting
+
+
+class FillObjects(cellprofiler.module.ImageProcessing):
+    category = "Advanced"
+
+    module_name = "FillObjects"
+
+    variable_revision_number = 1
+
+    def create_settings(self):
+        super(FillObjects, self).create_settings()
+
+        self.size = cellprofiler.setting.Float(
+            text="Minimum hole size",
+            value=64.,
+            doc="Holes smaller than this diameter will be filled."
+        )
+
+    def settings(self):
+        __settings__ = super(FillObjects, self).settings()
+
+        return __settings__ + [
+            self.size
+        ]
+
+    def visible_settings(self):
+        __settings__ = super(FillObjects, self).visible_settings()
+
+        return __settings__ + [
+            self.size
+        ]
+
+    def run(self, workspace):
+        self.function = lambda image, diameter: \
+            fill_object_holes(image, diameter)
+
+        super(FillObjects, self).run(workspace)
+
+
+def fill_object_holes(image, diameter):
```

Comment:
If you inherited from `cellprofiler.module.ObjectProcessing`, this `image` parameter would be a label matrix. It would have 2 dimensions or 3 dimensions, corresponding to 2D or 3D objects. You wouldn't have to check if the third dimension represented channels.

Reply:
Good to know!